sim-lib/src/lib.rs

```diff
@@ -418,16 +418,58 @@ async fn consume_simulation_results(
     shutdown: triggered::Trigger,
 ) {
     log::debug!("Simulation results consumer started.");
+    let mut result_logger = PaymentResultLogger::new();
 
     while let Some(resolved_payment) = receiver.recv().await {
-        // TODO - write to CSV.
-        println!("Resolved payment received: {:?}", resolved_payment);
+        result_logger.report_result(&resolved_payment.0, &resolved_payment.1);
+        log::trace!("Resolved payment received: {:?}", resolved_payment);
```

Comment:
For readability, I think it makes more sense to destructure `resolved_paymeny`

```suggestion
    while let Some((details, result)) = receiver.recv().await {
        result_logger.report_result(&details, &result);
        log::trace!("Resolved payment received: ({:?}, {:?})", details, result);
    }
```